edd-purchase-gravatars.php

```diff
@@ -328,28 +333,33 @@ public function register_widget() {
 		}
 
 		/**
-		 * Shortcode
+		 * Outputs the customer gravatars for a specific download.
 		 *
 		 * @since 1.0
-		 * @todo set the ID to get_the_ID() if ID parameter is not passed through. Otherwise it will incorrectly get other gravatars
-		*/
-		public function shortcode( $atts, $content = null ) {
+		 * @param array $atts The array of attributes passed to the shortcode.
+		 */
+		public function shortcode( $atts ) {
 
-			extract( shortcode_atts( array(
-					'id' => '',
-					'title' => ''
-				), $atts, 'edd_purchase_gravatars' )
+			$atts = shortcode_atts(
+				array(
+					'id'    => '',
+					'title' => '',
+				),
+				$atts,
+				'edd_purchase_gravatars'
 			);
 
-			// if no ID is passed on single download pags, get the correct ID
+			// On a singular download, get the current ID.
 			if ( is_singular( 'download' ) ) {
```

Comment:
```suggestion
			if ( is_singular( 'download' ) && empty( $atts['id'] ) ) {
```

I know this isn't new code, but based on the old comment, I feel like this condition should be in there? :thinking:

Reply:
That would be a behavior change--the current code always uses `get_the_ID()` on a singular download, whether or not an ID was passed through. If you're okay with the behavior change, I'll make it--otherwise I would suggest maybe updating the doc block to more clearly explain the behavior on singular downloads.

Reply:
Okay leave it as is for now. We'll come back to it later if we feel like it.